<a href="https://colab.research.google.com/github/gcompanys/OrganizacionDeDatos/blob/main/Problema1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd 
import numpy as np 

notas_df = pd.read_csv('/content/notasAlumnos.csv')
notas_df.sample(5)

padron      materia  nota     fecha
44   10333     analisis     6  30-04-20
48   10306       taller     8  31-12-19
87   10370  computacion     2  14-04-21
20   10308     analisis    10  29-07-20
46   10308   algoritmos     9  15-02-21

**A - Cuales es el promedio general de notas**

In [3]:
notas_df['nota'].mean()

5.35

Por lo tanto el promedio general de notas es aproximadamente 5.35


**B - Cual es la nota mas alta y la nota mas baja registrada durante el anio 2019**

In [7]:
#primero modifico la columna fecha y lo llevo a del tipo datetime
notas_df['fecha'] = notas_df['fecha'].apply(pd.to_datetime) 

In [49]:
notas_2019 = notas_df[notas_df['fecha'].dt.year == 2019] #importante usar el .dt.year
notas_aprobadas_2019 = notas_2019[notas_2019['nota'] >= 4]
notas_aprobadas_2019

padron  materia  nota      fecha
48   10306   taller     8 2019-12-31
51   10375  algebra     4 2019-11-11
72   10345   taller     8 2019-10-10
74   10311   fisica     5 2019-11-16

Tengo un dataframe con todas las notas con fecha den 2019

In [6]:
notaMax2019 = notas_2019['nota'].max()
notaMin2019 = notas_2019['nota'].min()

print('Nota mas alta en 2019: ', notaMax2019)
print('Nota mas baja en 2019: ', notaMin2019)

Nota mas alta en 2019:  8
Nota mas baja en 2019:  1


**C - Cual es el padron con mayor cantidad de materias aprobadas durante el ultimo cuatrimestre**

Cosas a tener en cuenta:


1.   Un alumno aprueba con 4
2.   Ultimo cuatrimestre va de 1/8/19 al 12/12/19



In [52]:
fechas_cuatrimestre = ('2019-08-01' <= notas_df['fecha']) & (notas_df['fecha'] <= '2019-12-12')
aprobados = notas_df['nota'] >= 4

aprobados_cuatrimestre = notas_df.loc[fechas_cuatrimestre & aprobados, 'padron']
aprobados_cuatrimestre.value_counts().nlargest()

10375    1
10345    1
10311    1
Name: padron, dtype: int64

En este caso tuve 3 padrones que aprobaron la mayor cantidad de materias en el ultimo cuatrimestre (aprobaron 1 sola materia)

**D - Cual es la nota promedio por materia**

In [60]:
notas_df.loc[:,['materia','nota']].groupby('materia').mean()

nota
materia              
algebra      5.368421
algoritmos   5.733333
analisis     5.928571
computacion  6.153846
fisica       4.727273
numerico     4.230769
taller       5.133333

**E - Cual es la nota promedio por padron**

In [61]:
#Una forma de hacerlo igual a la anterior
notas_df.loc[:, ['padron','nota']].groupby('padron').mean()

nota
padron      
10300   10.0
10301    1.0
10302    4.0
10303    8.5
10304    2.0
...      ...
10381    9.0
10382    5.0
10383    9.5
10384    3.0
10385    3.0

[61 rows x 1 columns]

In [64]:
#Otra forma de hacerlo
notas_df[['padron','nota']].groupby('padron').mean()

nota
padron      
10300   10.0
10301    1.0
10302    4.0
10303    8.5
10304    2.0
...      ...
10381    9.0
10382    5.0
10383    9.5
10384    3.0
10385    3.0

[61 rows x 1 columns]